internal/utils.go

```diff
@@ -1,32 +1,24 @@
 package internal
 
 import (
-	"math/rand"
+	"fmt"
 	"strings"
+
+	"github.com/codecrafters-io/tester-utils/random"
 )
 
-func repeatSlice(slice []string, n int) []string {
-	ret := make([]string, 0)
-	for i := 0; i < n; i++ {
-		ret = append(ret, slice...)
-	}
-	return ret
+func getRandInt() int {
+	return random.RandomInt(10, 100)
 }
 
-func randomSelection(count int, characters []string, joinWith string) string {
-	selectedCharacters := make([]string, count)
-	for i := 0; i < count; i++ {
-		selectedCharacters[i] = characters[rand.Intn(len(characters))]
-	}
-	return strings.Join(selectedCharacters, joinWith)
+func joinWith[T any](arr []T, sep string) string {
+	return strings.Join(convertToStringSlice(arr), sep)
```

Comment:
Interesting - I actually haven't gotten to using generics in Go yet, but does just doing this: 

```
strings.Join(random.RandomElementsFromArray(WHITESPACES, 2), "")
```

not work? I'd expect that since `WHITESPACES` is `[]string`, the resulting type from `RandomElementsFromArray` would be `[]string` 🤔

Reply:
Initially I had thought we would have more varied array types, where we would have to cast everything to string, this would have helped remove all the repeated casting logic. 
But as of now, we only use `[]string`, so removed `joinWith()`.